mempool/src/interface/mempool_interface.rs

```diff
@@ -23,17 +23,23 @@ use subsystem::{CallRequest, ShutdownRequest};
 
 pub trait MempoolInterface: Send + Sync {
     /// Add a transaction to mempool
-    fn add_transaction(&mut self, tx: SignedTransaction) -> Result<(), Error>;
+    fn add_transaction(&mut self, tx: SignedTransaction) -> Result<TxStatus, Error>;
 
     /// Get all transactions from mempool
     fn get_all(&self) -> Result<Vec<SignedTransaction>, Error>;
 
-    /// Get a specific transaction from the mempool
+    /// Get a specific transaction from the main mempool
```

Comment:
Is there a definition somewhere to what "main mempool" means? If you mean that this excludes orphans, please write it explicitly. Instead of main mempool, we say "from the mempool, excluding orphans".